### PR TITLE
Add preservation callback for items to push item ids into redis

### PR DESCRIPTION
**Pull request by murny**
Code to resolve first bullet in #702

### CODE REVIEW

File: app/models/concerns/item_properties.rb [hunk 1]
```diff
@@ -169,6 +170,34 @@ def purge_files
...
+        # TODO: <removed temporarily> log to external service iff result != true
+        true
+      rescue StandardError
+        # we trap errors in writing to the Redis queue in order to avoid crashing the save process for the user.
+        Rollbar.error("Error occured in push_item_id_for_preservation, Could not preserve #{id}", e)
+        true
+      end
+
+      # rubocop:disable Style/GlobalVars
+      def preserve
```
Comment (murny) [inline app/models/concerns/item_properties.rb]: Hydranorth has moved this out to a lib file: https://github.com/ualbertalib/HydraNorth/blob/master/lib/hydranorth/preservation_queue.rb

Which don't think we need to do here. Makes it much simiplier this way (no need to pass in items id etc)

File: app/models/concerns/item_properties.rb [hunk 2]
```diff
@@ -169,6 +170,34 @@ def purge_files
...
+        queue_name = Rails.application.secrets.preservation_queue_name
+
+        $queue ||= ConnectionPool.new(size: 1, timeout: 5) { Redis.current }
```
Comment (murny) [inline app/models/concerns/item_properties.rb]: I assume connection pool wants this to be a global variable for it to do the connection pooling stuff. So I disabled rubocop for this

File: app/models/concerns/item_properties.rb [hunk 3]
```diff
@@ -169,6 +170,34 @@ def purge_files
...
 
+      def push_item_id_for_preservation
+        result = preserve
+
+        Rails.logger.warn("Could not preserve #{id}") if result != true
+        # TODO: <removed temporarily> log to external service iff result != true
+        true
+      rescue StandardError
+        # we trap errors in writing to the Redis queue in order to avoid crashing the save process for the user.
+        Rollbar.error("Error occured in push_item_id_for_preservation, Could not preserve #{id}", e)
```
Comment (murny) [inline app/models/concerns/item_properties.rb]: There was a TODO in the old code for adding a rollbar notification? https://github.com/ualbertalib/HydraNorth/blob/master/app/models/concerns/hydranorth/generic_file/preservation.rb#L18

So added one here

File: app/models/concerns/item_properties.rb [hunk 4]
```diff
@@ -169,6 +170,34 @@ def purge_files
         self.ordered_members = []
       end
 
+      def push_item_id_for_preservation
+        result = preserve
+
+        Rails.logger.warn("Could not preserve #{id}") if result != true
+        # TODO: <removed temporarily> log to external service iff result != true
```
Comment (murny) [inline app/models/concerns/item_properties.rb]: Should we be logging this to rollbar as well?  Or logging this is fine?

File: lib/tasks/jupiter.rake [hunk 5]
```diff
@@ -1,10 +1,10 @@
-namespace :cleanup do
+namespace :jupiter do
```
Comment (murny) [inline lib/tasks/jupiter.rake]: This rake task was out growing it's "cleanup" name. Rename this file and namespace to be jupiter.

File: test/models/item_test.rb [hunk 6]
```diff
@@ -418,4 +418,139 @@ class ItemTest < ActiveSupport::TestCase
     end
   end
 
+  # Preservation queue handling
```
Comment (murny) [inline test/models/item_test.rb]: Brought over existing tests from hydranorth: https://github.com/ualbertalib/HydraNorth/blob/master/spec/models/generic_file_spec.rb#L217-L312

Converted them to minitest and using activesupport time helpers

File: test/test_helper.rb [hunk 7]
```diff
@@ -31,6 +31,7 @@
 
 class ActiveSupport::TestCase
 
+  # TODO: Remove after 5.2 PR gets merged
```
Comment (murny) [inline test/test_helper.rb]: This helper gets added in 5.2, so we can just remove ours once we finish upgrading

File: app/models/concerns/item_properties.rb (hunk 4 above)
Comment (mbarnett) [reply app/models/concerns/item_properties.rb]: Rollbar, probably? There's like zero monitoring of the Rails logs in production

Comment (murny) [reply app/models/concerns/item_properties.rb]: Okay adding this to the PR right now. Will need reapproval after.

**Pull request merged**